source/slang/slang-parser.cpp

```diff
@@ -235,9 +235,9 @@ namespace Slang
         {
             bindingToByteOffset.set(binding, byteOffset);
         }
-        int getNextBindingOffset(int binding) 
+        const int getNextBindingOffset(int binding) 
         {
-            int& currentOffset = bindingToByteOffset.getOrAddValue(binding, 0);
+            const int currentOffset = bindingToByteOffset.getOrAddValue(binding, 0);
```

Comment:
Because of the name of the function, I got an impression that this function doesn't mutate "this".
But the function "getOrAddValue()" will mutate when it add.
So you cannot apply "const" to the function.
What I suggest was
```
int getNextBindingOffset(int binding) const
```
But you cannot do it.
Adding "const" to the return type has no meaning when the type is a primitive type; please remove it.

Reply:
getNextBindingOffset mutates a dictionary, but the value returned is const, it should never be changed once returned, only used or discarded

Reply:
It seems that you are not vary familiar with how "const" modifier works.
I recommend you to checkout a fantastic book called, "Effective C++ - Scott Meyers" if you haven't.

Note that you cannot force the calling site to treat the return value as const as you intended.
There will be no error nor warning when you do something like,
```
const int getNextBindingOffset(int binding) { ...}
int a = getNextBindingOffset(...); // no warning no error.
```